src/stan/mcmc/base_mcmc.hpp

```diff
@@ -18,15 +18,15 @@ namespace stan {
 
       virtual sample
       transition(sample& init_sample,
-                 interface_callbacks::writer::base_writer& info_writer,
-                 interface_callbacks::writer::base_writer& error_writer) = 0;
+                 callbacks::writer& info_writer,
+                 callbacks::writer& error_writer) = 0;
 
       virtual void get_sampler_param_names(std::vector<std::string>& names) {}
 
       virtual void get_sampler_params(std::vector<double>& values) {}
 
       virtual void
-      write_sampler_state(interface_callbacks::writer::base_writer& writer) {}
+      write_sampler_state(callbacks::writer& writer) {}
```

Comment:
@bob-carpenter, that'll come later. I completely agree that we need sampler-state specific writers rather than overloading a generic writer. We'll get there, but after this pull request. I'm thinking about it.